src/bootman/bootman.c

```diff
@@ -446,6 +458,174 @@ bool boot_manager_set_uname(BootManager *self, const char *uname)
         return self->have_sys_kernel;
 }
 
+static int initrd_nodep_name_cmp(const void *file1, const void *file2)
+{
+        return strcmp((char*)file1, (char*)file2);
+}
+
+void boot_manager_get_initrd_nodep(BootManager *self)
+{
+        DIR *dir = NULL;
+        struct dirent *ent = NULL;
+        struct stat st = { 0 };
+        NcArray *initrd_nodep = NULL;
+        if (!self || !self->initrd_nodep_dir) {
+                return;
+        }
+
+        initrd_nodep = nc_array_new();
+        OOM_CHECK_RET(initrd_nodep,);
+
+        dir = opendir(self->initrd_nodep_dir);
+        if (!dir) {
+                LOG_ERROR("Error opening %s: %s", self->initrd_nodep_dir, strerror(errno));
+                nc_array_free(&initrd_nodep, NULL);
+                return;
+        }
+
+        while ((ent = readdir(dir)) != NULL) {
+        char *initrd_name = NULL;
+                autofree(char) *path = NULL;
+
+                path = string_printf("%s/%s", self->initrd_nodep_dir, ent->d_name);
+
+                if (strstr(ent->d_name, "img.gz\0") == NULL) {
```

Comment:
Seems a rather odd thing to hard-code the compression method and suffix in the private initrd directory.

Reply:
I think in a suffix in order to find freestanding initrd in the boot dir when they have to be deleted, I understand your comment, I will think in a name preffix instead